vector/src/main/java/im/vector/app/core/mvrx/ResultExtension.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.mvrx
+
+import com.airbnb.mvrx.Async
+import com.airbnb.mvrx.Fail
+import com.airbnb.mvrx.Success
+
+fun <A> Result<A>.foldToAsync(): Async<A> = fold(
```

Comment:
Seems this is always called after `runCatching`. Might be worth making a function that does both `runCatching` and `foldToAsync` in one go.

Reply:
Also, be aware that we are losing the Loading state here. FTR, Mavericks 2 has full coroutine support and already provide methods for this case.

Reply:
Thanks, I've handled your remarks.